### ISSUE TITLE
Activity extraction failed - Azure Automation Runbook Import

### ISSUE THREAD

**Issue by kduraira** [dfinke/ImportExcel#201]
Just tried to import this module (version: 3.0.1) into my Azure Runbook and package extraction seems to fail consistently? At first, I attempted to import from the gallery from within Azure Portal but that failed. Secondly, I downloaded and installed the module from PSGallery, I zipped up the folder from my PS Modules path and imported that on Azure and that failed as well. The error is consistent on both occurrences:

> Error extracting the activities from module ImportExcel. Extraction failed with the following error: Orchestrator.Shared.AsyncModuleImport.ModuleImportException: While importing the module, an error occurred while extracting activities from the module. Internal error message: Unable to find type [PSPlot]..

Any suggestions on how I can go about this would be awesome. Thanks.

**Comment by dfinke** [comment]
Thanks for the issue. What version of PowerShell is running in the Azure Runbook? `$PSVersionTable`

**Comment by kduraira** [comment]
Hi Doug, thanks for getting back. Commenting out the above lines worked, thanks. But when MS auto-updates installed modules, this will fail again. Is there something I can do so it won't break my runbooks without any manual intervention at all? Thanks again.

**Comment by dfinke** [comment]
Please, could you try commenting out just the two lines with the write-warning. If that works, I'll comment them out in the gallery version and that should work

**Comment by kduraira** [comment]
It fails to extract again with the same error:

> Error extracting the activities from module ImportExcel. Extraction failed with the following error: Orchestrator.Shared.AsyncModuleImport.ModuleImportException: While importing the module, an error occurred while extracting activities from the module. Internal error message: Unable to find type [PSPlot]..

**Comment by dfinke** [comment]
Thanks for trying. I'll comment out all of those lines and publish that to the gallery. 

One last request, please run `$PSVersionTable` and past the results. 

Thank you

**Comment by glaisne** [comment]
Thought I'd add my 2 cents. I came across this yesterday. Looking closer at the error, I commented out [OutputType([PSPlot])] (line 30) (V 3.0.0), zipped and uploaded it without an issue. I'm doing some small tests now (Azure Hybrid automation).
My guess is there is a race condition in the loading of the module, where the class hasn't gotten created yet before the properties of the function are set.

Here is the $PSVersionTable (run in Azure not in hybrid on-prem system).
Name                                     Key                                     Value                                  
----                                     ---                                     -----                                  

PSVersion                                PSVersion                               5.1.14409.1005                         
PSEdition                                PSEdition                               Desktop                                
PSCompatibleVersions                     PSCompatibleVersions                    {1.0, 2.0, 3.0, 4.0...}                
BuildVersion                             BuildVersion                            10.0.14409.1005                        
CLRVersion                               CLRVersion                              4.0.30319.19455                        
WSManStackVersion                        WSManStackVersion                       3.0                                    
PSRemotingProtocolVersion                PSRemotingProtocolVersion               2.3                                    
SerializationVersion                     SerializationVersion                    1.1.0.1
